app/views/handlers/block.py

```diff
@@ -135,24 +136,26 @@ def _update_section_completeness(
             list_item_id=location.list_item_id,
         )
 
-    def _set_started_at_metadata(self):
+    def _set_started_at_metadata(self) -> str | None:
         response_metadata = self._questionnaire_store.response_metadata
         if not response_metadata.get("started_at"):
             started_at = datetime.now(timezone.utc).isoformat()
             logger.info("Survey started", started_at=started_at)
             response_metadata["started_at"] = started_at
 
-    def _get_safe_page_title(self, page_title):
+    def _get_safe_page_title(self, page_title: str) -> str:
         page_title = self._schema.get_single_string_value(page_title)
         return safe_content(page_title)
 
-    def _resolve_custom_page_title_vars(self) -> MutableMapping:
+    def _resolve_custom_page_title_vars(self) -> dict[str, int]:
         list_item_position = self._questionnaire_store.list_store.list_item_position(
-            self.current_location.list_name, self.current_location.list_item_id
+            # Type ignore: no list_item_id or list_name is available yet
+            self.current_location.list_name,
```

Comment:
This comment doesn't quite make sense, do you mean other way round? you may need an ignore above as well